### PR TITLE
avoids displaying key and trust store passwords

### PR DESCRIPTION
**Pull request by shamsimam**
## Changes proposed in this PR

- avoids displaying key and trust store passwords

## Why are we making these changes?

We wish to avoid displaying passwords in our apis.

Screen Shot 2019-03-22 at 4 15 47 PM

### CODE REVIEW

**Comment by sradack** [comment]
Why aren't we using the `#config/env` reader tag?

**Comment by shamsimam** [comment]
@sradack I also don't want them to show up in the settings endpoint. With this change, I do not have to worry about remembering to delete the password entries before display in the handler.

File: waiter/src/waiter/settings.clj [hunk 1]
```diff
@@ -214,22 +214,29 @@
...
+      (if-not (and (map? data) (contains? data :kind))
+        data
+        (->> (keys data)
+             (remove #(let [nested-data (get data %)]
+                        (and (not= % :kind)
+                             (not= % (:kind data))
+                             (map? nested-data)
+                             (contains? nested-data :factory-fn))))
+             (select-keys data))))
+    (cond-> (utils/dissoc-in settings [:zookeeper :connect-string])
```
Comment (sradack) [inline waiter/src/waiter/settings.clj]: Why are we `dissoc`ing some and using `_hidden_` with others?

Comment (shamsimam) [reply waiter/src/waiter/settings.clj]: Done. Use `<hidden>` consistently.

**Comment by shamsimam** [comment]
@sradack ready for review.

**Pull request merged**